Automattic/wp-calypso
# display the total clicks and total impressions in advertising pages

Related to #

## Proposed Changes

2638-gh-tumblr/a8c-dsp

## Why are these changes being made?
we want to add a total impressions and total clicks (for all campaigns) for a user in the advertising pages (both in ready to promote and campaigns page)

the added elements look like this (ss from browser)
<img width="400" alt="Screenshot 2024-10-15 at 18 15 32" src="https://github.com/user-attachments/assets/190a1c77-1941-4bea-a13f-cb1e631ad62a">

and ss from figma 
<img width="645" alt="Screenshot 2024-10-15 at 18 17 11" src="https://github.com/user-attachments/assets/1adb8db6-ffa8-4e06-a60f-6a0477b593d4">

please note that I haven't added the icons for more info (the ones with the 'i') but I am missing the context of what will happen if hovered ( I guess we need some kind of popover with some additional messages)

## Update:
I added a function to format large numbers properly..

as we have users who might have very large numbers of campaigns and their impressions (maybe also clicks) can be added to have millions of results 

now for example 100000 will be displayed as 100K 125500 will be displayed as 125.5K 

1000000 will be displayed as 1M 

1255000 will be displayed as 1.255M and so on (for millions I added 3 decimals to have some better understanding of the amount)

to test this you can just hardcode the line 

`totalClicks={ campaignsStats?.total_clicks }`

to
`totalClicks={ 1500000 }`
in

client/my-sites/promote-post-i2/main.tsx

 
*

## Testing Instructions

please check out the branch (or test the preview links) check that the elements appear as expected
(I checked the layout both when we do have a banner and when we don't.. you can also test by hardcoding this line
`const showBanner = ! campaignsIsLoading && ( totalCampaignsUnfiltered || 0 ) < 3;` and change it to showBanner = true or false accordingly

test that the layout looks ok in tablet and mobile..

do a CR.

## Pre-merge Checklist
